crates/shared/src/order_validation.rs

```diff
@@ -506,6 +503,59 @@ impl OrderValidating for OrderValidator {
     }
 }
 
+/// Order validity period configuration.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct OrderValidityConfiguration {
+    pub min: Duration,
+    pub max_market: Duration,
+    pub max_limit: Duration,
+}
+
+impl OrderValidityConfiguration {
+    /// Validates an order's timestamp based on additional data.
+    fn validate_period(&self, order: &PreOrderData) -> Result<(), OrderValidityError> {
+        let now = model::time::now_in_epoch_seconds();
+        if order.valid_to < timestamp_after_duration(now, self.min) {
+            return Err(OrderValidityError::InsufficientValidTo);
+        }
+        if let Some(max) = self.max(order) {
+            if order.valid_to > timestamp_after_duration(now, max) {
+                return Err(OrderValidityError::ExcessiveValidTo);
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Returns the maximum valid timestamp for the specified order.
+    fn max(&self, order: &PreOrderData) -> Option<Duration> {
+        // For now, there is no maximum `validTo` for pre-sign orders as a hack
+        // for dealing with signature collection times. We should probably
+        // revisit this.
+        if order.signing_scheme == SigningScheme::PreSign {
+            return None;
+        }
+
+        match order.class {
+            OrderClass::Ordinary => Some(self.max_market),
+            OrderClass::Limit => Some(self.max_limit),
+            OrderClass::Liquidity => None,
+        }
+    }
+}
+
+/// Adds a `std::time::Duration` to a `u32` timestamp. This function uses
+/// saturating semantics and can't panic.
+fn timestamp_after_duration(timestamp: u32, duration: Duration) -> u32 {
+    timestamp.saturating_add(duration.as_secs().try_into().unwrap_or(u32::MAX))
+}
```

Comment:
nit: I feel this should be moved also to `model::time` so it could be reused. There are a few more places where we add duration to u32.

Reply:
Good call - didn't notice we had a shared module for this kind of thing.

Reply:
Implemented in df4d722b06517a166f0df3e0f2653990c6810204.